src/vcpkg/tools.cpp

```diff
@@ -56,17 +114,34 @@ namespace vcpkg
         parsed_version.normalize();
 
         auto d1 = Strings::strto<int>(parsed_version.major);
-        if (!d1.has_value()) return {};
+        if (!d1.has_value()) return msg::format(msgToolVersionMetadataIllFormed, msg::actual = string_version);
 
         auto d2 = Strings::strto<int>(parsed_version.minor);
-        if (!d2.has_value()) return {};
+        if (!d2.has_value()) return msg::format(msgToolVersionMetadataIllFormed, msg::actual = string_version);
 
         auto d3 = Strings::strto<int>(parsed_version.patch);
-        if (!d3.has_value()) return {};
-
-        return std::array<int, 3>{*d1.get(), *d2.get(), *d3.get()};
+        if (!d3.has_value()) return msg::format(msgToolVersionMetadataIllFormed, msg::actual = string_version);
```

Comment:
I think this would be cleaner if reduced to a single check

```cpp
if (!d1.has_value() || !d2.has_value() || !d3.has_value()) /*error*/
```

at the end